packages/hardhat-test-helpers/src/utils.ts

```diff
@@ -0,0 +1,88 @@
+import type { EIP1193Provider } from "hardhat/types";
+
+import type { NumberLike } from "./types";
+
+async function checkIfHardhatNetwork(
+  provider: EIP1193Provider
+): Promise<boolean> {
+  try {
+    // we call a method that only exists in the hardhat network
+    await provider.request({
+      method: "hardhat_getAutomine",
```

Comment:
I used this method because I was being lazy, but we should use `web3_clientVersion` instead and parse the result.

We should also handle the case where this method doesn't exist, just in case. If it doesn't, then it's not a Hardhat Network and we just return false too.